vm-systemd/network-uplink-setup.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+# Source Qubes library.
+# shellcheck source=init/functions
+. /usr/lib/qubes/init/functions
+
+# Setup IP address at specific time of system boot, instead of asynchronously
+# by udev
+QUBES_MANAGED_IFACE="$(get_qubes_managed_iface)"
+if [ "x$QUBES_MANAGED_IFACE" != "x" ]; then
+    INTERFACE="$QUBES_MANAGED_IFACE" ACTION="add" /usr/lib/qubes/setup-ip
+fi
+
+# only now allow udev to (re)configure the interface - prevent enabling it
+# earlier, which would mess systemd ordering
+touch /run/qubes-network-uplink-enable
```

Comment:
This can be dealt with by having `systemd-udevd` enqueue a start job, which respects ordering.